modulus/models/unet/unet3d.py

```diff
@@ -0,0 +1,510 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import List, Optional, Union
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from ..meta import ModelMetaData
+from ..module import Module
+
+
+class ConvBlock(nn.Module):
```

Comment:
Can the layers from here be reused? https://github.com/NVIDIA/modulus/blob/main/modulus/models/rnn/layers.py

I believe we have implementation of ConvBlock, TransposeConvBlock and ResidualBlock. Alternatively, can we use this PR to unify them?

Reply:
Well, not exactly. The conv block in this PR also includes normalization and activation, which is not there in the rnn layers. I agree that we have different implementation of these layers, and that eventually these have to be unified. For now, hopefully, the api documentation helps to distinguish between these layers/modules.

Reply:
Sounds good. Let's make an issue for this, so that it can be tracked in the future

Reply:
Sure, I added it to this list: https://github.com/NVIDIA/modulus/issues/463